xu69ux/graphiql-app
# Environment Setup

**Title:** Environment Setup

**Description:** This pull request includes the necessary configuration changes for setting up the development environment. The changes include:

- Updated `tsconfig.json` to include `"noImplicitAny"`: false to allow implicit any types.
- Added a `.prettierrc` file to enforce consistent code formatting rules across the project.
- Updated the fix script in package.json to run `Prettier` on `.js, .jsx, .ts, and .tsx` files in the src directory.
- Installed `eslint-config-prettier` to resolve conflicts between Prettier and ESLint rules.
- Set up `Jest` for unit testing, including the necessary configuration in `jest.config.js`.
- Installed and configured `Husky` to enforce code quality standards before commits and pushes.
- These changes will help ensure consistent code style, reduce potential errors in the project, and enforce code quality standards through testing and pre-commit checks.

**Test Plan:**

1. Run `npm run fix` and ensure that `Prettier` formats the code without errors.
2. Check that `ESLint` does not report any conflicts with `Prettier`.
3. Run `npm test` to ensure Jest runs the tests without errors.
4. Make a commit to ensure `Husky` runs the `pre-commit` hooks without issues.
